artemis-server/src/test/java/org/apache/activemq/artemis/core/config/impl/FileConfigurationBrokerConnectionEncryptedTest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.core.config.impl;
+
+import java.util.List;
+
+import org.apache.activemq.artemis.core.config.Configuration;
+import org.apache.activemq.artemis.core.config.FileDeploymentManager;
+import org.apache.activemq.artemis.core.config.amqpBrokerConnectivity.AMQPBrokerConnectConfiguration;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class FileConfigurationBrokerConnectionEncryptedTest extends ConfigurationImplTest {
+
+   protected String getConfigurationName() {
+      return "ConfigurationTest-broker-connection-encrypted-config.xml";
+   }
+
+   @Override
+   @Test
+   public void testDefaults() {
+      // empty
+   }
+
+   @Test
+   public void testAMQPBrokerConfigEncryptedUserAndPassword() {
+
+      List<AMQPBrokerConnectConfiguration> brokerConnections = conf.getAMQPConnection();
+      Assert.assertNotNull("brokerConnections is not null", brokerConnections);
+      Assert.assertTrue("brokerConnections is not empty", !brokerConnections.isEmpty());
```

Comment:
The assertion message should describe what's wrong if the condition isn't met
```suggestion
      Assert.assertNotNull("brokerConnections is null", brokerConnections);
      Assert.assertFalse("brokerConnections is empty", brokerConnections.isEmpty());
```

Reply:
Assertion messages stating the opposite of the assertion logic always seemed backwards to me, but that's also one of those perennial "spaces vs. tabs" debates.